src/App.tsx

```diff
@@ -1,27 +1,84 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum Sort {
+  None = '',
+  All = 'all-button',
+  FirstFive = 'first-five-buttons',
+  Red = 'red-buttons',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sort, setSort] = useState<Sort>(Sort.None);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleClick = useCallback((sortType: Sort) => {
+    if (sort === sortType) {
+      return;
+    }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    switch (sortType) {
+      case Sort.All:
+        getAll()
+          .then(setGoods);
+        break;
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      case Sort.FirstFive:
+        get5First()
+          .then(setGoods);
+        break;
+
+      case Sort.Red:
+        getRedGoods()
+          .then(setGoods);
+        break;
+
+      default:
+        break;
+    }
+
+    setSort(sortType);
+  }, []);
```

Comment:
The useCallback dependency array is empty, but it should include 'sort' as a dependency. Update the array to `[sort]` to ensure the callback updates when 'sort' changes. 🚀